saw-central/src/SAWCentral/Crucible/MIR/TypeShape.hs

```diff
@@ -376,28 +382,90 @@ fieldShapeMirTy :: FieldShape tp -> M.Ty
 fieldShapeMirTy (ReqField shp) = shapeMirTy shp
 fieldShapeMirTy (OptField shp) = shapeMirTy shp
 
+-- | This is to accomodate multiple Rust types mapping to the same Cryptol
+-- type.  For example, if a Cryptol function expects [3][8], we could map
+-- it to a Rust function that either expects `[u8;3]`, or `&[u8]` with a
+-- dynamic check that it has 3 elements.  The type parameter `a` is for the
+-- the lengths of the slices---it will be either Cryptol's `Type` during
+-- type checking, or `Intgeger`, once we instantiate a schema at a concrete
+-- type.
+data CryTermAdaptor a =
+    NoAdapt                               -- ^ Use default translation
+  | AdaptTuple [CryTermAdaptor a]         -- ^ Adapt a tuple
+  | AdaptArray (CryTermAdaptor a)         -- ^ Adapt an array
+  | AdaptDerefSlice M.Collection a (CryTermAdaptor a)
+    -- ^ A reference to a slice.
+    -- We also store the collection here so that we can convert MIR types
+    -- to typeshapes when we need to.
```

Comment:
It might be worth clarifying that you want to convert the slice's element type specifically:

```suggestion
    -- We also store the 'M.Collection' here so that we can convert the
    -- element type's 'TypeRepr' to a 'TypeShape' when we need to (see,
    -- for instance, the implementation of `shapeToTerm'`).
```